src/pymor/operators/numpy.py

```diff
@@ -324,10 +326,18 @@ def apply_inverse(self, V, mu=None, initial_guess=None, least_squares=False,
                     raise InversionError(f'{str(type(e))}: {str(e)}') from e
                 R = R.T
             else:
-                try:
-                    R = solve(self.matrix, V.to_numpy().T).T
-                except np.linalg.LinAlgError as e:
-                    raise InversionError(f'{str(type(e))}: {str(e)}') from e
+                if not hasattr(self, '_lu_factor'):
+                    try:
+                        self._lu_factor = lu_factor(self.matrix)
+                    except np.linalg.LinAlgError as e:
+                        raise InversionError(f'{str(type(e))}: {str(e)}') from e
+                    gecon = get_lapack_funcs('gecon', self._lu_factor)
+                    rcond, _ = gecon(self._lu_factor[0], np.linalg.norm(self.matrix, ord=1), norm='1')
+                    if rcond < np.finfo(np.float64).eps:
+                        warn('Ill-conditioned matrix (rcond={:.6g}): '
+                             'result may not be accurate.'.format(rcond),
+                             LinAlgWarning, stacklevel=3)
```

Comment:
Any reason you are using `warn` instead of `logger.warn` here?

Reply:
I just copy-pasted this from SciPy. I'm not sure which type of warning is more appropriate.

Reply:
@sdrave Thoughts on this? I guess `warn` is good since it's the same in SciPy, but `logging.warning` is also good because the condition number check after `lu_solve` is specific to pyMOR.

Reply:
Sounds reasonable. I made the change now.

Reply:
Thx, sounds good. Marking this as resolved, another detailed question to follow elsewhere...